chainntnfs/bitcoindnotify/bitcoind.go

```diff
@@ -385,60 +390,127 @@ out:
 // historicalConfDetails looks up whether a transaction is already included in a
 // block in the active chain and, if so, returns details about the confirmation.
 func (b *BitcoindNotifier) historicalConfDetails(txid *chainhash.Hash,
+	heightHint, currentHeight uint32) (*chainntnfs.TxConfirmation, error) {
+
+	// First, we'll attempt to retrieve the transaction details using the
+	// backend node's transaction index.
+	txConf, err := b.confDetailsFromTxIndex(txid)
+	if err != nil {
+		return nil, err
+	}
+
+	if txConf != nil {
+		return txConf, nil
+	}
+
+	// If the backend node's transaction index is not enabled, then we'll
+	// fall back to manually scanning the chain's blocks, looking for the
+	// block where the transaction was included in.
+	return b.confDetailsManually(txid, heightHint, currentHeight)
+}
+
+// confDetailsFromTxIndex looks up whether a transaction is already included
+// in a block in the active chain by using the backend node's transaction index.
+// If it is found, the confirmation details are returned. Otherwise, an error
```

Comment:
Comment describing if/when an error is returned is confusing.

Reply:
Fixed.